File: lib/core/marketing/interfaces/marketing_service.dart
import 'package:sertton/core/global/responses/rest_response.dart';
import 'package:sertton/core/marketing/dtos/banner_dto.dart';
import 'package:sertton/core/marketing/dtos/lead_dto.dart';

abstract class MarketingService {
  Future<RestResponse<List<BannerDto>>> fetchBanners();
  Future<RestResponse<void>> saveLead(LeadDto leadDto);
}

File: src/core/marketing/interfaces/marketing-service.ts
import type { Banner, Lead } from "@/core/marketing/entities"
import type { RestResponse } from "@/core/shared/responses/rest-response"

export interface MarketingService {
  fetchBanners(): Promise<RestResponse<readonly Banner[]>>
  saveLead(lead: Lead): Promise<RestResponse<void>>
}

export type IMarketingService = MarketingService
